File: IT/1/Variabili, operatori, strutture di controllo/Esercizi/Python/esercizio24.py
# file esercizio24.py

from __future__ import print_function

if hasattr(__builtins__, 'raw_input'): input = raw_input

s = input("Scrivi un numero: ")

if not s.isdigit():
	print("Numero non valido!")
	exit()

n = int(s)

ex = 2
while (ex <= 10):
	print("Potenza di {} elevato a {} = {}".format(n, ex, n**ex))
	ex += 1

File: IT/1/Variabili, operatori, strutture di controllo/Esercizi/C++/esercizio24.cpp
/* 
file esercizio24.cpp
compilare con: g++ -std=c++11 esercizio24.cpp
*/

#include <iostream>
#include <cmath>

using std::cin;
using std::cout;
using std::endl;
using std::string;

int main() 
{
	cout << "Scrivi un numero: ";

	string s;
	getline(cin, s);

	if (s.find_first_not_of("0123456789") != string::npos) {
		cout << "Numero non valido!" << endl;
		return 0;
	}

	int n = stoi(s);

	int ex = 2;

	while (ex <= 10) {
		cout << "Potenza di " << n << " elevato a " << ex << " = " << pow(n, ex) << endl;
		ex++;
	}
}

